docs/src/main/asciidoc/spring-cloud-gateway.adoc

```diff
@@ -1358,32 +1478,159 @@ spring:
   cloud:
     gateway:
       routes:
-      - id: rewriteresponseheader_route
+      - id: requestratelimiter_route
         uri: https://example.org
         filters:
-        - RewriteResponseHeader=X-Response-Red, , password=[^&]+, password=***
+        - name: RequestRateLimiter
+          args:
+            redis-rate-limiter.replenishRate: 10
+            redis-rate-limiter.burstCapacity: 20
+            redis-rate-limiter.requestedTokens: 1
+
 ----
 ====
 
-For a header value of `/42?user=ford&password=omg!what&flag=true`, it is set to `/42?user=ford&password=\***&flag=true` after making the downstream request.
-You must use `$\` to mean `$` because of the YAML specification.
-
-=== The `SaveSession` `GatewayFilter` Factory
-
-The `SaveSession` `GatewayFilter` factory forces a `WebSession::save` operation _before_ forwarding the call downstream.
-This is of particular use when using something like https://projects.spring.io/spring-session/[Spring Session] with a lazy data store and you need to ensure the session state has been saved before making the forwarded call.
-The following example configures a `SaveSession` `GatewayFilter`:
+The following example configures a KeyResolver in Java:
 
-.application.yml
+.Config.java
 ====
-[source,yaml]
+[source,java]
 ----
-spring:
-  cloud:
-    gateway:
-      routes:
-      - id: save_session
-        uri: https://example.org
+@Bean
+KeyResolver userKeyResolver() {
+    return exchange -> Mono.just(exchange.getRequest().getQueryParams().getFirst("user"));
+}
+----
+====
+
+This defines a request rate limit of 10 per user. A burst of 20 is allowed, but, in the next second, only 10 requests are available.
+The `KeyResolver` is a simple one that gets the `user` request parameter (note that this is not recommended for production).
```

Comment:
Promote the note from parentheses to a caution on a separate line. Things that aren't recommended for production should always be clearly called out.